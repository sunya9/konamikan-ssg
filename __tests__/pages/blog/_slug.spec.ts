import { mount } from '@vue/test-utils'
import { PostOrPage } from '@tryghost/content-api'
import { post } from '@/__tests__/fixtures/postOrPage'
import Article from '@/pages/blog/_slug.vue'

describe('article page', () => {
  it('normal', () => {
    const newerPost: PostOrPage = { ...post, title: 'newerPost' }
    const olderPost: PostOrPage = { ...post, title: 'olderPost' }
    expect(
      mount(Article, {
        mocks: {
          post,
          newerPost,
          olderPost
        }
      })
    ).toMatchSnapshot()
  })
})
