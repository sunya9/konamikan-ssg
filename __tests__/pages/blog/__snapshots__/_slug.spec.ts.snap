// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`article page normal 1`] = `
<div>
  <header class="hero is-medium fix-background" style="background-image: url(cover.png);">
    <div class="hero-head">
      <nav class="navbar is-transparent">
        <div class="container">
          <div class="navbar-brand">
            <div class="navbar-brand"><a class="navbar-item has-text-white-bis">
                title
              </a></div> <a href="#" aria-label="Menu" class="navbar-burger has-text-white"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
          </div>
          <div class="navbar-menu is-relative">
            <div class="navbar-end">
              <div class="navbar-item">
                <div class="control has-icons-right"><label for="search-form-input" class="is-sr-only">Search</label> <input id="search-form-input" type="search" placeholder="Search..." autocomplete="off" class="input is-small search"> <span class="icon is-small is-right"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"><circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line></svg></span></div>
                <!---->
              </div> <a href="/rss" class="navbar-item has-text-white-ter"><span class="icon"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-rss"><path d="M4 11a9 9 0 0 1 9 9"></path><path d="M4 4a16 16 0 0 1 16 16"></path><circle cx="5" cy="19" r="1"></circle></svg></span> <span class="is-sr-only">RSS</span></a>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div class="hero-body fix-z-index">
      <div class="container has-text-centered">
        <h1 class="title is-1 has-text-white">
          title
        </h1>
        <!---->
        <div class="container">
          <div class="level">
            <div class="level-item has-text-centered content"><span class="tag is-rounded is-medium">
              2020/04/05 12:34:56
            </span></div>
          </div>
        </div>
      </div>
    </div>
    <!---->
  </header>
  <main class="section">
    <div class="container">
      <article>
        <div class="content">
          <div>
            <p>article</p>
          </div>
        </div>
        <footer>
          <div class="level is-mobile">
            <div class="level-left">
              <div class="level-item">
                <div class="tags"><a class="tag is-primary is-light">
                    tag
                  </a></div>
              </div>
            </div>
            <div class="level-right">
              <div class="level-item">
                <div class="dropdown is-right is-active">
                  <div class="dropdown-trigger"><button class="share-button">
                      <div aria-haspopup="" class="icon"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-share-2">
                          <circle cx="18" cy="5" r="3"></circle>
                          <circle cx="6" cy="12" r="3"></circle>
                          <circle cx="18" cy="19" r="3"></circle>
                          <line x1="8.59" y1="13.51" x2="15.42" y2="17.49"></line>
                          <line x1="15.41" y1="6.51" x2="8.59" y2="10.49"></line>
                        </svg></div>
                    </button></div>
                  <!---->
                </div>
              </div>
            </div>
          </div>
          <div class="columns is-centered">
            <div class="column is-10">
              <div class="box">
                <div class="media">
                  <div class="media-left">
                    <div class="image is-64x64"><a><img src="" width="128" height="128" loading="lazy" class="is-rounded"></a></div>
                  </div>
                  <div class="media-content">
                    <div class="content is-small">
                      <div class="title is-6"><a>
                          Tester
                        </a></div>
                      <div class="subtitle is-6">
                        <!---->
                      </div>
                      <p></p>
                      <div class="level is-mobile">
                        <div class="level-left">
                          <!---->
                          <!---->
                          <!---->
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <nav aria-label="この記事の前後の記事">
            <ul class="columns is-vcentered is-mobile">
              <li class="column has-text-left"><a aria-label="新しい記事" class="is-block">
                  <div class="vertical-center"><span class="icon"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-left"><polyline points="15 18 9 12 15 6"></polyline></svg></span> <span>newerPost</span></div>
                </a></li>
              <li class="column has-text-right"><a aria-label="古い記事" class="is-block">
                  <div class="vertical-center"><span>olderPost</span> <span class="icon"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-chevron-right"><polyline points="9 18 15 12 9 6"></polyline></svg></span></div>
                </a></li>
            </ul>
          </nav>
        </footer>
      </article>
    </div>
  </main>
</div>
`;
